src/gt4py/caching.py

```diff
@@ -154,6 +156,10 @@ def class_name(self) -> str:
         """Calculate the name for the stencil class, default is to read from build options."""
         return self.builder.options.name
 
+    def capture_externals(self) -> Dict[str, Any]:
+        """Extract externals from the annotated stencil definition for fingerprinting. Freezes the references."""
```

Comment:
I would replace "Freezes the references" with something like "Returns a snapshot of the externals when called".